File: scripts/translate.py
#!/usr/bin/env python3

import sys
import json
import deepl

if len(sys.argv) < 3 or not sys.argv[-2].isdigit():
    print('Usage: translate.py NUM_TO_TRANSLATE IN_FILE.json')
    sys.exit(1)

articles = []

with open(sys.argv[-1], 'r') as f:
    articles = json.load(f)
    todo = [article for article in articles if not 'translated_text' in article and article['is_confirmed'] is None]
    num = int(sys.argv[-2])
    for i in range(0, num):
        print('%s (%s/%s)' % (todo[i]['title'], i + 1, num))
        try:
            todo[i]['translated_text'] = deepl.translate(todo[i]['text'], target='DE')[0]
            todo[i]['translated_title'] = deepl.translate(todo[i]['title'], target='DE')[0]
            print('\t=> %s' % (todo[i]['translated_title']))
        except KeyError:
            print('Likely ran out of request quota! Aborting and saving all results thus far.')
            break

with open(sys.argv[-1], 'w') as f:
    f.write(json.dumps(articles, indent=2))

File: scripts/translate.js
const { translate } = require('deepl-translator')
const fs = require('fs-extra')

const max = parseInt(process.argv[process.argv.length - 2])
const filename = process.argv[process.argv.length - 1]

if (process.argv.length < 3 || isNaN(max)) {
  console.log('Usage: translate.py NUM_TO_TRANSLATE IN_FILE.json')
  process.exit(1)
}

(async () => {
  let text = await fs.readFile(filename)
  let data = JSON.parse(text)
  let count = 0

  for (let article of data) {
    if (article.translated_text) continue
    console.log(`${article.title} (${count + 1}/${max})`)
    article.translated_text = (await translate(article.text, 'DE')).translation
    article.translated_title = (await translate(article.title, 'DE')).translation
    console.log(`\t=> ${article.translated_title}`)
    if (++count === max) break
  }

  fs.writeFile(filename, JSON.stringify(data, null, '  '))
})().catch(err => { throw err })
